### PR TITLE
fix src/rabbit_auth_backend_cache.erl:21: undefined callback function…

### PR DESCRIPTION
**Pull request by folt**
fix crash install "src/rabbit_auth_backend_cache.erl:21: undefined callback function check_topic_access/4 (behaviour 'rabbit_authz_backend')"

### CODE REVIEW

**Comment by michaelklishin** [comment]
Sorry but this is very light on details. This plugin *is* supposed to act as an authorization backend.

The problem is that you are trying to build it from master, whereas `check_topic_access/4` is only available in RabbitMQ master/3.7.0 and not 3.6.x.

Build from the stable branch.

File: src/rabbit_auth_backend_cache.erl [hunk 1]
```diff
@@ -18,7 +18,6 @@
 -include_lib("rabbit_common/include/rabbit.hrl").
 
 -behaviour(rabbit_authn_backend).
--behaviour(rabbit_authz_backend).
```
Comment (michaelklishin) [inline src/rabbit_auth_backend_cache.erl:21]: This plugin *is* supposed to provide an implementation of `rabbit_authz_backend`.

**Comment by folt** [comment]
I use RabbitMQ 3.6.9 
At me after operation sudo make
The script fails with an error "src/rabbit_auth_backend_cache.erl:21: undefined callback function check_topic_access/4 (behaviour 'rabbit_authz_backend')"

After studying the file, it turned out that there were 2 identical calls
Removing one of them solved the problem and everything works

**Comment by michaelklishin** [comment]
You are not mentioning what branch of this plugin you are building it from and that is critically important here.

I'm going to produce a binary build against the tip of stable (identical to `3.6.9` in this repo) anyhow. There should no reason to build this plugin from source.

**Comment by michaelklishin** [comment]
@folt I'm sorry but you are confused and you don't realise it. There are no "identical calls". The lines you are referring to are

``` erlang
-behaviour(rabbit_authn_backend).
-behaviour(rabbit_authz_backend).
```

which ARE NOT identical. It is easy to verify using GitHub search and searching for `rabbit_authn_backend` (which is the authentication module behaviour) and `rabbit_authz_backend` (the authorization module behaviour) individually.

Consider understanding [the difference between them](https://github.com/rabbitmq/internals/blob/master/authorization_and_authentication.md) before you suggest that we remove one of them as "a duplicate".

**Comment by folt** [comment]
I tried all the branches
The plugin is not going to

**Comment by michaelklishin** [comment]
Right, so your solution is to delete a line without understanding why it's there.

I'm going to upload a binary build of this plugin shortly. End of story.

**Comment by folt** [comment]
@michaelklishin 
Really they are very similar
This is confusing

**Comment by michaelklishin** [comment]
@folt here's [an updated binary build](https://github.com/rabbitmq/rabbitmq-auth-backend-cache/releases/tag/rabbitmq_v3_6_9). Please give it a shot.

**Comment by folt** [comment]
@michaelklishin  thanks for the help

**Comment by michaelklishin** [comment]
We are considering including this plugin into a future RabbitMQ release. Might as well consider it for 3.6.10 since it's fairly popular.

**Comment by folt** [comment]
@michaelklishin If it does not complicate you, then it would be nice to make an example for setting it up with http authentication :-)

**Comment by michaelklishin** [comment]
This is not a support venue. There are [examples available](https://github.com/rabbitmq/rabbitmq-auth-backend-cache/tree/stable#authentication-and-authorization-backend-configuration) in the README. In case you have specific questions, please post them to rabbitmq-users.

**Pull request closed without merging**